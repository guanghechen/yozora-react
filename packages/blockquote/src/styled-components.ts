import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
