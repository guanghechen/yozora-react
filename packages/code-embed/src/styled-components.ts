import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
