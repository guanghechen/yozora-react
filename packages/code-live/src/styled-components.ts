import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
