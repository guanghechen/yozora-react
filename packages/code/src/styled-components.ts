import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
