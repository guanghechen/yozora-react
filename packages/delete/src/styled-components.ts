import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
