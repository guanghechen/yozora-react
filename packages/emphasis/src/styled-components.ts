import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
