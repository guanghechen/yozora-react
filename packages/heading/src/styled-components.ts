import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
