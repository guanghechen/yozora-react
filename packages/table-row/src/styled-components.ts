import type { YozoraTableRowTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    tableRow?: YozoraTableRowTheme
  }
}
