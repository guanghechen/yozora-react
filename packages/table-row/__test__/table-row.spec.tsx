import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import TableRow from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a simple content', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <table>
        <TableRow>
          <th><span>{ text }</span></th>
        </TableRow>
      </table>
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-list-item'
    const wrapper = render(
      <table>
        <TableRow>
          <td className={ className }><span>{ text }</span></td>
        </TableRow>
      </table>
    )
    expect(wrapper.find('.' + className) != null).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('children is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(
          <table>
            <TableRow>{ value }</TableRow>
          </table>
        )
      }).toThrow(/The prop `children` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLTableRowElement>()
    const wrapper = mount(
      <table>
        <tbody>
          <TableRow ref={ ref } data-value="waw">
            <td>1</td>
          </TableRow>
        </tbody>
      </table>
    )

    const o = wrapper.find('tr').getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <table>
        <tbody>
          <TableRow style={{ color: 'orange', fontSize: '16px' }}>
            <td>
              some text1
              <span>some text2</span>
            </td>
          </TableRow>
        </tbody>
      </table>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        tableRow: {
          // background: 'red',
          evenBackground: 'blue',
        }
      }
    }

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <table>
          <tbody>
            <TableRow style={{ color: 'orange', fontSize: '16px' }}>
              <td>
                some text1
                <span>some text2</span>
              </td>
            </TableRow>
          </tbody>
        </table>
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
