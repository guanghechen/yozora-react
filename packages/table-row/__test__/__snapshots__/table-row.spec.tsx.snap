// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  background: none;
}

.c0:nth-child(2n) {
  background: none;
}

<table>
  <tbody>
    <tr
      class="c0"
      style="color:orange;font-size:16px"
    >
      <td>
        some text1
        <span>
          some text2
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c0 {
  background: none;
}

.c0:nth-child(2n) {
  background: blue;
}

<Component
  theme={
    Object {
      "yozora": Object {
        "tableRow": Object {
          "evenBackground": "blue",
        },
      },
    }
  }
>
  <table>
    <tbody>
      <YozoraTableRow
        style={
          Object {
            "color": "orange",
            "fontSize": "16px",
          }
        }
      >
        <styled.tr
          style={
            Object {
              "color": "orange",
              "fontSize": "16px",
            }
          }
          theme={
            Object {
              "yozora": Object {
                "tableRow": Object {
                  "background": "none",
                  "evenBackground": "none",
                },
              },
            }
          }
        >
          <tr
            className="c0"
            style={
              Object {
                "color": "orange",
                "fontSize": "16px",
              }
            }
          >
            <td>
              some text1
              <span>
                some text2
              </span>
            </td>
          </tr>
        </styled.tr>
      </YozoraTableRow>
    </tbody>
  </table>
</Component>
`;
